### PR TITLE
Capture and Settings API updates

### PR DESCRIPTION
**Pull request by torbsorb**
This PR focus on updates required to support SDK 2.0 API for Capture and Settings.

### CODE REVIEW

File: source/applications/advanced/color_balance.py [hunk 1]
```diff
@@ -26,6 +26,11 @@ class MeanColor:
     blue: np.array
 
 
+def _convert_frame2d_to_numpy_float_rgb(frame: zivid.Frame2D):
```
Comment (SatjaSivcev) [inline source/applications/advanced/color_balance.py:29]: name too detailed? Maybe better frame2d_to_rgb? frame2d_to_numpy_rgb?

File: source/applications/advanced/hand_eye_calibration/calibrate_eye_to_hand.py [hunk 2]
```diff
@@ -7,14 +7,25 @@
...
+        acquisitions=[
+            zivid.Settings.Acquisition(
+                aperture=8, exposure_time=datetime.timedelta(microseconds=20000),
+            ),
+        ],
+        processing=zivid.Settings.Processing(
+            filters=zivid.Settings.Processing.Filters(
+                smoothing=zivid.Settings.Processing.Filters.Smoothing(
+                    gaussian=zivid.Settings.Processing.Filters.Smoothing.Gaussian(
+                        enabled=True
```
Comment (SatjaSivcev) [inline source/applications/advanced/hand_eye_calibration/calibrate_eye_to_hand.py:21]: space in front of and after '='

File: source/applications/advanced/gamma_correction.py [hunk 3]
```diff
@@ -61,7 +61,7 @@ def _capture_2d_image(camera):
         OpenCV BGR image
 
     """
-    settings_2d = zivid.Settings2D()
+    settings_2d = zivid.Settings2D(acquisitions=[zivid.Settings2D.Acquisition()],)
```
Comment (SatjaSivcev) [inline source/applications/advanced/gamma_correction.py:64]: space before and after "="?

File: source/applications/advanced/hand_eye_calibration/ur_hand_eye_calibration/universal_robots_perform_hand_eye_calibration.py [hunk 4]
```diff
@@ -170,19 +173,31 @@ def _get_frame_and_transform_matrix(con: rtde, cam: zivid.Camera):
...
+                exposure_time=datetime.timedelta(microseconds=10000),
+                brightness=1.0,
+                gain=1,
+            )
+        ],
+        processing=zivid.Settings.Processing(
+            filters=zivid.Settings.Processing.Filters(
+                smoothing=zivid.Settings.Processing.Filters.Smoothing(
+                    gaussian=zivid.Settings.Processing.Filters.Smoothing.Gaussian(
+                        enabled=True
```
Comment (SatjaSivcev) [inline source/applications/advanced/hand_eye_calibration/ur_hand_eye_calibration/universal_robots_perform_hand_eye_calibration.py:195]: spaces, "="?

File: source/applications/advanced/hand_eye_calibration/ur_hand_eye_calibration/universal_robots_perform_hand_eye_calibration.py [hunk 5]
```diff
@@ -170,19 +173,31 @@ def _get_frame_and_transform_matrix(con: rtde, cam: zivid.Camera):
...
-        updater.settings.iris = 17
-        updater.settings.exposure_time = datetime.timedelta(microseconds=10000)
-        updater.settings.filters.reflection.enabled = True
-        updater.settings.brightness = 1.0
-        updater.settings.filters.gaussian.enabled = 1
-        updater.settings.gain = 1
+    return zivid.Settings(
+        acquisitions=[
+            zivid.Settings.Acquisition(
+                aperture=7.98,
```
Comment (SatjaSivcev) [inline source/applications/advanced/hand_eye_calibration/ur_hand_eye_calibration/universal_robots_perform_hand_eye_calibration.py:185]: use 'real' f-numbers, so 8.0

File: source/camera/basic/CaptureTutorial.md [hunk 6]
```diff
@@ -47,7 +46,7 @@ camera = app.connect_camera(serial_number="2020C0DE")
 ---
 **Note** 
 
-The serial number of your camera is shown in the Zivid Studio.
+The serial number of your camera is shown in Zivid Studio.
```
Comment (SatjaSivcev) [inline source/camera/basic/CaptureTutorial.md:49]: should this be part of a different commit?

File: source/camera/basic/CaptureTutorial.md [hunk 7]
```diff
@@ -111,68 +108,73 @@ We may choose to configure settings manually. For more information about what ea
...
+    processing=zivid.Settings.Processing(
+        filters=zivid.Settings.Processing.Filters(
+            outlier=zivid.Settings.Processing.Filters.Outlier(
+                removal=zivid.Settings.Processing.Filters.Outlier.Removal(
+                    enabled=True, threshold=5
+                )
+            )
+        )
+    ),
+)
```
Comment (SatjaSivcev) [inline source/camera/basic/CaptureTutorial.md:126]: spaces ' = '?

File: source/camera/basic/CaptureTutorial.md [hunk 8]
```diff
@@ -111,68 +108,73 @@ We may choose to configure settings manually. For more information about what ea
...
 
-It is possible to only capture a 2D image. This is faster than a 3D capture, and can be used . 2D settings are configured as follows ([go to source][settings2d-url]).
+It is possible to only capture a 2D image. This is faster than a 3D capture. 2D settings are configured as follows ([go to source][settings2d-url]).
 ```python
-settings_2d = zivid.Settings2D()
-settings_2d.exposure_time = datetime.timedelta(microseconds=10000)
-settings_2d.gain = 1
-settings_2d.iris = 35
-settings_2d.brightness = 1
+settings = zivid.Settings2D(
```
Comment (SatjaSivcev) [inline source/camera/basic/CaptureTutorial.md:149]: spaces ' = '

File: source/camera/basic/CaptureTutorial.md [hunk 9]
```diff
@@ -111,68 +108,73 @@ We may choose to configure settings manually. For more information about what ea
...
 
-We may also set a list of settings to be used in an [HDR capture](#capture-hdr) ([go to source][settingsHDR-url]).
+We may also create settings to be used in an HDR capture ([go to source][settings-hdr-url]).
 ```python
-settings_list = [camera.settings for _ in range(3)]
-settings_list[0].iris = 14
-settings_list[1].iris = 21
-settings_list[2].iris = 35
+settings = zivid.Settings(
+    acquisitions=[
```
Comment (SatjaSivcev) [inline source/camera/basic/CaptureTutorial.md:134]: spaces ' = '

File: source/camera/basic/capture.py [hunk 10]
```diff
@@ -7,12 +7,15 @@ def _main():
     app = zivid.Application()
     camera = app.connect_camera()
 
-    with camera.update_settings() as updater:
-        updater.settings.iris = 40
-        updater.settings.exposure_time = datetime.timedelta(microseconds=40000)
-        updater.settings.filters.reflection.enabled = True
+    settings = zivid.Settings(
```
Comment (SatjaSivcev) [inline source/camera/basic/capture.py:10]: spaces ' = '

File: source/camera/basic/capture_2d.py [hunk 11]
```diff
@@ -7,12 +7,16 @@ def _main():
     app = zivid.Application()
     camera = app.connect_camera()
 
-    settings_2d = zivid.Settings2D()
-    settings_2d.iris = 50
-    settings_2d.exposure_time = datetime.timedelta(microseconds=50000)
+    settings_2d = zivid.Settings2D(
+        acquisitions=[
```
Comment (SatjaSivcev) [inline source/camera/basic/capture_2d.py:11]: space ' = '

File: source/camera/basic/capture_from_file.py [hunk 12]
```diff
@@ -12,7 +12,15 @@ def _main():
         Path() / get_sample_data_path() / "FileCameraZividOne.zfc"
     )
 
-    with camera.capture() as frame:
+    settings = zivid.Settings(
+        acquisitions=[
```
Comment (SatjaSivcev) [inline source/camera/basic/capture_from_file.py:16]: space '  = '

File: source/camera/basic/capture_hdr.py [hunk 13]
```diff
@@ -6,12 +6,15 @@ def _main():
...
 
-    settings_list = [camera.settings for _ in range(3)]
-    settings_list[0].iris = 14
-    settings_list[1].iris = 21
-    settings_list[2].iris = 35
+    settings = zivid.Settings(
+        acquisitions=[
+            zivid.Settings.Acquisition(aperture=10.9,),
+            zivid.Settings.Acquisition(aperture=5.8,),
+            zivid.Settings.Acquisition(aperture=2.8,),
```
Comment (SatjaSivcev) [inline source/camera/basic/capture_hdr.py:13]: 11.31, 5.66, 2.83

File: source/camera/basic/capture.py [hunk 14]
```diff
@@ -7,12 +7,15 @@ def _main():
...
+        acquisitions=[
+            zivid.Settings.Acquisition(
+                aperture=5.6, exposure_time=datetime.timedelta(microseconds=8333),
```
Comment (SatjaSivcev) [inline source/camera/basic/capture.py:13]: 5.66

File: source/camera/basic/CaptureTutorial.md [hunk 15]
```diff
@@ -111,68 +108,73 @@ We may choose to configure settings manually. For more information about what ea
...
-    updater.setting.filters.outlier.enabled = True
-    updater.setting.filters.outlier.threshold = 5
-    updater.setting.filters.reflection.enabled = True
-    updater.setting.filters.saturated.enabled = True
-    updater.setting.blue_balance = 1.081
-    updater.setting.red_balance = 1.709
+settings = zivid.Settings(
+    acquisitions=[
+        zivid.Settings.Acquisition(
+            aperture=5.6, exposure_time=datetime.timedelta(microseconds=8333),
```
Comment (SatjaSivcev) [inline source/camera/basic/CaptureTutorial.md:114]: 5.66

File: source/applications/basic/capture_hdr_separate_frames.py [hunk 16]
```diff
@@ -1,62 +0,0 @@
-"""
```
Comment (SatjaSivcev) [inline source/applications/basic/capture_hdr_separate_frames.py:1]: Commit message should explain why this is removed

File: source/camera/basic/capture_hdr_complete_settings.py [hunk 17]
```diff
@@ -17,58 +17,47 @@ def _main():
...
-    settings_collection[2].filters.outlier.enabled = True
-    settings_collection[2].filters.outlier.threshold = 5
-    settings_collection[2].filters.reflection.enabled = True
-    settings_collection[2].filters.saturated.enabled = True
-    settings_collection[2].blue_balance = 1.081
-    settings_collection[2].red_balance = 1.709
+    settings = zivid.Settings(
+        acquisitions=[
+            zivid.Settings.Acquisition(
+                aperture=23.42,
```
Comment (SatjaSivcev) [inline source/camera/basic/capture_hdr_complete_settings.py:24]: https://github.com/zivid/zivid-cpp-samples/blob/master/source/Camera/Basic/CaptureHDRCompleteSettings/CaptureHDRCompleteSettings.cpp

Use same settings

File: source/utils/settings_from_file.py [hunk 18]
```diff
@@ -0,0 +1,79 @@
...
+
+
+def get_settings_from_yaml(path: Path) -> zivid.Settings:
+    """Get settings from .yml file.
+
+    Args:
+        path: Path to .yml file which contains settings
+
+    Returns:
+        settings: a Zivid Settings object
```
Comment (SatjaSivcev) [inline source/utils/settings_from_file.py:68]: Zivid settings

File: source/utils/settings_from_file.py [hunk 19]
```diff
@@ -0,0 +1,79 @@
...
+
+    """
+    settings_from_yaml = yaml.load(path.read_text(), Loader=yaml.Loader)["Settings"]
+    settings = zivid.Settings(
+        acquisitions=_acquisitions_from_yaml(settings_from_yaml["Acquisitions"]),
+        processing=zivid.Settings.Processing(
```
Comment (SatjaSivcev) [inline source/utils/settings_from_file.py:74]: space ' = '

File: source/applications/advanced/color_balance.py (hunk 1 above)
Comment (SatjaSivcev) [reply source/applications/advanced/color_balance.py]: done

File: source/applications/advanced/hand_eye_calibration/calibrate_eye_to_hand.py (hunk 2 above)
Comment (SatjaSivcev) [reply source/applications/advanced/hand_eye_calibration/calibrate_eye_to_hand.py:21]: will leave to the linter to decide

File: source/applications/advanced/gamma_correction.py (hunk 3 above)
Comment (SatjaSivcev) [reply source/applications/advanced/gamma_correction.py:64]: will leave to the linter to decide

File: source/applications/advanced/hand_eye_calibration/ur_hand_eye_calibration/universal_robots_perform_hand_eye_calibration.py (hunk 4 above)
Comment (SatjaSivcev) [reply source/applications/advanced/hand_eye_calibration/ur_hand_eye_calibration/universal_robots_perform_hand_eye_calibration.py:195]: will leave to the linter to decide

File: source/applications/advanced/hand_eye_calibration/ur_hand_eye_calibration/universal_robots_perform_hand_eye_calibration.py (hunk 5 above)
Comment (SatjaSivcev) [reply source/applications/advanced/hand_eye_calibration/ur_hand_eye_calibration/universal_robots_perform_hand_eye_calibration.py]: done, here and for other samples

**Comment by SatjaSivcev** [comment]
Fixed my own comments. Someone else should review.

File: source/camera/basic/CaptureTutorial.md (hunk 6 above)
Comment (torbsorb) [reply source/camera/basic/CaptureTutorial.md:49]: maybe, but I wouldn't want a separate commit for it in the master history. It will be squashed after PR anyway, so maybe just leave it?
